adr/ADR-8.md

```diff
@@ -302,58 +321,30 @@ Any received messages that isn't a Delete/Purge operation gets added to the list
-or a codec.
-
-Read only clients would only need a decoder - and so by extension, not need a private key.
-
-All keys and values should be encoded/decoded. If users wish to encrypt their buckets both the keys and the values will be encrypted.
-The valid keys after encryption should still match the pattern for acceptable keys. Base64 encoding encrypted keys is a good option.
-
-Encoding keys is done on a per token basis, the key `hello.world` would be encoded in parts, for example a Base64 encoding of this would
-be `aGVsbG8=.d29ybGQ=` wildcards are not encoded. This means if one watch the key `hello.>` it would encode to `aGVsbG8=.>` and still
-do the right thing with regard to watches and ranges.
-
-### Local Read Caches
-
-In a super-cluster the Bucket can be far away and have some latency, a read-cache is implemented that wraps the JetStream backend.
-
-The cache is warmed up using a Bucket Watch, the cache is ready when it's reached the `pending=0` message from the watch or when the
-nil message indicating no data is received.
-
-While the cache is still warming up it's a pass through to the underlying backend for any reads.
-
-Most operations are pass-through, except:
+Whatchers should support at least the following options. Languages can choose to support more models if they wish, as long as that
+is clearly indicated as a language specific extension. Names should be language idiomatic but close to these below.
 
- * `Get()` will read local cache if ready (fully warmed), and a key is there
- * `Delete()` and `Purge()` will evict the key from local cache and pass through
- * `Put()` will evict the key from local cache and pass through
- * `Destroy()` will delete all messages from local cache and pass through
- * `Keys()` can be served immediately out of the local cache map
+We need to signal when we reach the end of the initial data set to facilitate use cases such as dumping a bucket, iterating keys etc.
+Languages can implement an End Of Initial Data signal in a language idiomatic manner. Internal to the watcher you reach this state the
+first time any message has a `Pending==0`. This signal must also be sent if no data is present - either by checking for messages using 
+`GetLastMsg()` on the watcher range or by inspecting the Pending+Delivered after creating the consumer. The signal must always be sent.
 
-We specifically do not place the `Put()` value into local cache since an Entry must know the JetStream sequence.
+|Name|Description|
+|----|-----------|
+|`IncludeHistory`|Send all available history rather than just the latest entries|
+|`IgnoreDeletes`|Only sends `PUT` operation entries|
+|`MetaOnly`|Does not send any values, only metadata about those values|
+|`NewUpdatesOnly`|Only sends new updates made, no current or historical values are sent. The End Of Initial Data marker is sent as soon as the watch starts.|
```

Comment:
Feedback on the name please, I know @jnmoyne had some suggestions but I missplaced them...

Reply:
@jnmoyne has posted this in the chat:
> newValuesOnly() or updatesOnly()

Reply:
+1 `updatesOnly`

Reply:
+1 on `updatesOnly`